sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/GroupIntoBatchesTest.java

```diff
@@ -116,16 +123,6 @@ public void testInGlobalWindowBatchSizeCount() {
     PAssert.that("Incorrect batch size in one or more elements", collection)
         .satisfies(
             new SerializableFunction<Iterable<KV<String, Iterable<String>>>, Void>() {
-
-              private boolean checkBatchSizes(Iterable<KV<String, Iterable<String>>> listToCheck) {
-                for (KV<String, Iterable<String>> element : listToCheck) {
-                  if (Iterables.size(element.getValue()) != BATCH_SIZE) {
```

Comment:
I did notice that it's  `!=` and not `>` here, but the test is still valid with `>` (we have 10 elements, and 5 batch size, so it can't be anything but 5, and we check the batch count at the end with `EVEN_NUM_ELEMENTS / BATCH_SIZE`)

Reply:
I would say that this previous test was too strict and your update makes sense. GroupIntoBatches ensures that the batches aren't bigger than BATCH_SIZE elements.

Unfortunately I think the GroupIntoBatches specification is too loose since it uses words like `Aim to create batches`. It would be great if we could make it a strict guarantee, for example batches will never be bigger then element count, or that they will never be bigger then byte size (except for the case where a single element is bigger then byte size and it will show up in its own group). I wouldn't try to solve this here but it would make sense to have a bug/and or follow-up PR to make this explicit.